### PR TITLE
[maintenance] Fix cpp typos

### PR DESCRIPTION
**Pull request by NathanBaulch**
Continuation of #37541 but focused on C++.

### CODE REVIEW

**Review by yashykt — APPROVED** [review]

**Review by eugeneo — CHANGES_REQUESTED** [review]
Nathan, thank you for your contribution. It is looking great and I would like to ask you to revert one change. It touches a field on seemingly unused structure but we have clients that use gRPC in different ways and it could be it is needed for some project that maintains a slightly modified gRPC version so this change would break them.

Thanks!

File: src/cpp/ext/gcp/environment_autodetect.cc [hunk 1]
```diff
@@ -165,7 +165,7 @@ class EnvironmentAutoDetectHelper
  private:
   struct Attribute {
     std::string resource_attribute;
-    std::string metadata_server_atttribute;
+    std::string metadata_server_attribute;
```
[the remark this answers is not in the document]
Comment (NathanBaulch) [reply src/cpp/ext/gcp/environment_autodetect.cc:168]: Sure, what about the comment [further down](https://github.com/NathanBaulch/grpc/blob/typos-cpp/src/cpp/ext/gcp/environment_autodetect.cc#L306), which was already spelled correctly?

[the remark this answers is not in the document]
Comment (eugeneo) [reply src/cpp/ext/gcp/environment_autodetect.cc:168]: IMHO, we need to remove the structure. But please leave it unchanged until then.

**Review by eugeneo — APPROVED** [review]
Thanks!

**Comment by eugeneo** [comment]
@NathanBaulch there should be a pull request sent to you in your repo by a bot that fixes formatting and such.

**Comment by NathanBaulch** [comment]
Nothing so far @eugeneo 🤷

**Comment by eugeneo** [comment]
This is what linter is suggesting, annoying, I know ;) :
```
Failure
--- /var/local/git/grpc/test/cpp/end2end/xds/xds_server.h	2024-09-24 22:43:53.307127329 +0000
+++ /tmp/tmp.YOc80xpqnE	2024-09-24 22:50:10.174890286 +0000
@@ -371,7 +371,7 @@
       }
       if (check_version_callback_ != nullptr) {
         check_version_callback_(request.type_url(),
-                               client_resource_type_version);
+                                client_resource_type_version);
       }
     } else {
       int client_nonce;
--- /var/local/git/grpc/test/cpp/qps/qps_json_driver.cc	2024-09-24 22:43:53.323128337 +0000
+++ /tmp/tmp.UD577jO5i8	2024-09-24 22:50:21.515605925 +0000
@@ -104,8 +104,9 @@
     }
     size_t comma = next_entry.find(',');
     if (comma == std::string::npos) {
-      LOG(ERROR) << "Expected --per_worker_credential_types to be a list of the "
-                    "form: 'addr1,cred_type1;addr2,cred_type2;...' into.";
+      LOG(ERROR)
+          << "Expected --per_worker_credential_types to be a list of the "
+             "form: 'addr1,cred_type1;addr2,cred_type2;...' into.";
       abort();
     }
     std::string addr = next_entry.substr(0, comma);
```

**Review by yashykt — APPROVED** [review]

**Pull request closed without merging**